### PR TITLE
Ford safety: curvature error limit

### PR DESCRIPTION
**Pull request by sshane**
By asserting a curvature command close to the current curvature (from yaw rate and two speed signals), we can limit torque wind up if openpilot were to command a high curvature far away from current

### CODE REVIEW

File: board/safety.h [hunk 1]
```diff
@@ -409,7 +409,18 @@ bool max_limit_check(int val, const int MAX_VAL, const int MIN_VAL) {
   return (val > MAX_VAL) || (val < MIN_VAL);
 }
 
-// check that commanded value isn't too far from measured
+// check that commanded angle value isn't too far from measured
+bool angle_dist_to_meas_check(int val, struct sample_t *val_meas, const int MAX_ERROR, const int MAX_VAL) {
+
+  // val must always be near val_meas, limited the maximum value
```
Comment (sshane) [inline board/safety.h:415]:
```suggestion
  // val must always be near val_meas, limited to the maximum value
```

File: board/safety.h [hunk 2]
```diff
@@ -409,7 +409,19 @@ bool max_limit_check(int val, const int MAX_VAL, const int MIN_VAL) {
...
 }
 
-// check that commanded value isn't too far from measured
+// check that commanded angle value isn't too far from measured
+bool angle_dist_to_meas_check(int val, struct sample_t *val_meas, const int MAX_ERROR, const int MAX_VAL) {
+
+  // val must always be near val_meas, limited to the maximum value
+  // add 1 to not false trigger the violation
+  int highest_allowed = MIN(MAX(val_meas->max + MAX_ERROR + 1, -MAX_VAL), MAX_VAL);
+  int lowest_allowed = MIN(MAX(val_meas->min - MAX_ERROR - 1, -MAX_VAL), MAX_VAL);
```
Comment (adeebshihadeh) [inline board/safety.h:418]: use the new clamp macro https://github.com/commaai/panda/pull/1356

Comment (sshane) [reply board/safety.h:418]: 🙌

File: tests/safety/test_ford.py [hunk 3]
```diff
@@ -69,6 +80,11 @@ def setUp(self):
     self.safety.set_safety_hooks(Panda.SAFETY_FORD, 0)
     self.safety.init_tests()
 
+  def _curvature_meas_msg_array(self, curvature, speed):
```
Comment (adeebshihadeh) [inline tests/safety/test_ford.py:83]: i'd call this something like reset_curvature_measurements

**Review by adeebshihadeh — COMMENTED** [review]
Let's also double check the speed with another signal

File: tests/safety/test_ford.py (hunk 3 above)
Comment (sshane) [reply tests/safety/test_ford.py:83]: just copied the convention for angle

**Pull request merged**